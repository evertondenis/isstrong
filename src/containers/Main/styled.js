import styled from 'styled-components'

export const StyledMain = styled.main`
  display: flex;
  width: 100%;
  height: 100vh;

  .form__create-count {
    margin-top: 30px;
  }

  .loading{
    width: 50px;
  }

  .is-invalid {
    border-color: #f79682;
  }
`

export const StyledBox = styled.div`
  margin: auto;
  padding: 60px 70px 23px 70px;
  width: 515px;
  background: #fff;
  border: 3px solid #f2f2f2;
  box-sizing: border-box;

  h1 {
    font-size: 22px;
    color: #312f4f;
    text-align: center;
  }

  .box__msg {
    font-size: 1rem;
    color: #696d8c;
    text-align: center;
  }

  .box__img {
    margin: 10px;
  }
`
