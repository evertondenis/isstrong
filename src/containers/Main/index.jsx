import React, { Component, Fragment } from 'react'
import { Logo, Input, Password, Button } from 'components'
import { Checked, Loading } from 'core/assets/images'
import { StyledMain, StyledBox } from './styled'

class Main extends Component {
  state = {
    name: '',
    email: '',
    password: '',
    passwordValid: false,
    validEmail: '',
    validForm: false,
    sendDisabled: true,
    sending: false
  }

  onChange = () => {
    this.setState({
      name: this.refs.inputRefName.refs.inputRef.value,
      email: this.validationEmail(this.refs.inputRefEmail.refs.inputRef.value)
    }, this.enabledButton)
  }

  validationEmail = email => {
    const valid = email.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i)
    const validEmail = (!valid && email !== '') ? 'is-invalid' : ''
    this.setState({ validEmail })
    return valid ? email : 'invalid'
  }

  checkPassword = (valid, value) => {
    this.setState({
      password: value,
      passwordValid: valid,
    }, this.enabledButton)
  }

  enabledButton = () => {
    const { name, email, passwordValid } = this.state
    if (name !== '' && email !== 'invalid' && passwordValid) {
      return this.setState({ sendDisabled: false })
    }
    return this.setState({ sendDisabled: true })
  }

  createCount = data => {
    const { name, email, password } = this.state
    data.preventDefault()
    this.setState({ validForm: true })
    console.log('form: ', name, email, password)
  }

  renderLabel = () => {
    const { sending } = this.state
    return !sending ? 'Criar conta' : <img src={Loading} className="loading" alt="enviando"/>
  }

  render() {
    const { validForm, validEmail, sendDisabled } = this.state
    return (
      <StyledMain>
        <StyledBox>
          {/* <Logo logo={Brand} alt="Olist logo" /> */}

          {validForm ? (
            <Fragment>
              <Logo className="box__img" logo={Checked} alt="Cadastro efetuado" />
              <h1>Tudo certo</h1>
              <div className="box__msg">
                <p>Verifique sua caixa de entrada</p>
                <p>para confirmar seu e-mail</p>
              </div>
            </Fragment>
          ) : (
            <Fragment>
              <h1>Crie sua conta</h1>
              <form className="form__create-count">
                <Input
                  focus
                  label="Nome completo"
                  ref="inputRefName"
                  onChange={this.onChange}
                  required
                />
                <Input
                  className={validEmail}
                  label="E-mail"
                  ref="inputRefEmail"
                  onChange={this.onChange}
                  required
                />
                <Password
                  confirmPwd
                  passwordValid={this.checkPassword}
                />
                <Button
                  type="submit"
                  form="createCount"
                  onClick={data => this.createCount(data)}
                  disabled={sendDisabled}
                >
                  {this.renderLabel()}
                </Button>
              </form>
            </Fragment>
          )}
        </StyledBox>
      </StyledMain>
    )
  }
}

export default Main
