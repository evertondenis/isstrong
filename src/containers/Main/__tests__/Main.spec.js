import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import Main from '../'
import 'setupTest'

describe('Main.', () => {
  it('Should be render <Main />.', () => {
    const wrapper = shallow(<Main />)
    expect(wrapper).toMatchSnapshot()
  })
})
