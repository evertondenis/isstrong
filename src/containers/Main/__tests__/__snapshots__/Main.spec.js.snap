// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main. Should be render <Main />. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <Logo
                alt="Olist logo"
                className=""
                logo="logo.svg"
        />
        <Unknown>
                <h1>
                        Crie sua conta
                </h1>
                <form
                        className="form__create-count"
                >
                        <Input
                                className=""
                                focus={true}
                                label="Nome completo"
                                onChange={[Function]}
                                placeholder=""
                                required={true}
                                type="text"
                        />
                        <Input
                                className=""
                                focus={false}
                                label="E-mail"
                                onChange={[Function]}
                                placeholder=""
                                required={true}
                                type="text"
                        />
                        <Password
                                confirmPwd={true}
                                max={16}
                                min={6}
                                passwordValid={[Function]}
                                placeholder=""
                        />
                        <Button
                                className=""
                                disabled={true}
                                form="createCount"
                                onClick={[Function]}
                                type="submit"
                        >
                                Criar conta
                        </Button>
                </form>
        </Unknown>
</styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Logo
            alt="Olist logo"
            className=""
            logo="logo.svg"
/>,
          <Unknown>
            <h1>
                        Crie sua conta
            </h1>
            <form
                        className="form__create-count"
            >
                        <Input
                                    className=""
                                    focus={true}
                                    label="Nome completo"
                                    onChange={[Function]}
                                    placeholder=""
                                    required={true}
                                    type="text"
                        />
                        <Input
                                    className=""
                                    focus={false}
                                    label="E-mail"
                                    onChange={[Function]}
                                    placeholder=""
                                    required={true}
                                    type="text"
                        />
                        <Password
                                    confirmPwd={true}
                                    max={16}
                                    min={6}
                                    passwordValid={[Function]}
                                    placeholder=""
                        />
                        <Button
                                    className=""
                                    disabled={true}
                                    form="createCount"
                                    onClick={[Function]}
                                    type="submit"
                        >
                                    Criar conta
                        </Button>
            </form>
</Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "Olist logo",
            "className": "",
            "logo": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h1>
                Crie sua conta
</h1>,
              <form
                className="form__create-count"
>
                <Input
                                className=""
                                focus={true}
                                label="Nome completo"
                                onChange={[Function]}
                                placeholder=""
                                required={true}
                                type="text"
                />
                <Input
                                className=""
                                focus={false}
                                label="E-mail"
                                onChange={[Function]}
                                placeholder=""
                                required={true}
                                type="text"
                />
                <Password
                                confirmPwd={true}
                                max={16}
                                min={6}
                                passwordValid={[Function]}
                                placeholder=""
                />
                <Button
                                className=""
                                disabled={true}
                                form="createCount"
                                onClick={[Function]}
                                type="submit"
                >
                                Criar conta
                </Button>
</form>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Crie sua conta",
              },
              "ref": null,
              "rendered": "Crie sua conta",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Input
                    className=""
                    focus={true}
                    label="Nome completo"
                    onChange={[Function]}
                    placeholder=""
                    required={true}
                    type="text"
/>,
                  <Input
                    className=""
                    focus={false}
                    label="E-mail"
                    onChange={[Function]}
                    placeholder=""
                    required={true}
                    type="text"
/>,
                  <Password
                    confirmPwd={true}
                    max={16}
                    min={6}
                    passwordValid={[Function]}
                    placeholder=""
/>,
                  <Button
                    className=""
                    disabled={true}
                    form="createCount"
                    onClick={[Function]}
                    type="submit"
>
                    Criar conta
</Button>,
                ],
                "className": "form__create-count",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "",
                    "focus": true,
                    "label": "Nome completo",
                    "onChange": [Function],
                    "placeholder": "",
                    "required": true,
                    "type": "text",
                  },
                  "ref": "inputRefName",
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "",
                    "focus": false,
                    "label": "E-mail",
                    "onChange": [Function],
                    "placeholder": "",
                    "required": true,
                    "type": "text",
                  },
                  "ref": "inputRefEmail",
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "confirmPwd": true,
                    "max": 16,
                    "min": 6,
                    "passwordValid": [Function],
                    "placeholder": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Criar conta",
                    "className": "",
                    "disabled": true,
                    "form": "createCount",
                    "onClick": [Function],
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Criar conta",
                  "type": [Function],
                },
              ],
              "type": "form",
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <Logo
                    alt="Olist logo"
                    className=""
                    logo="logo.svg"
          />
          <Unknown>
                    <h1>
                              Crie sua conta
                    </h1>
                    <form
                              className="form__create-count"
                    >
                              <Input
                                        className=""
                                        focus={true}
                                        label="Nome completo"
                                        onChange={[Function]}
                                        placeholder=""
                                        required={true}
                                        type="text"
                              />
                              <Input
                                        className=""
                                        focus={false}
                                        label="E-mail"
                                        onChange={[Function]}
                                        placeholder=""
                                        required={true}
                                        type="text"
                              />
                              <Password
                                        confirmPwd={true}
                                        max={16}
                                        min={6}
                                        passwordValid={[Function]}
                                        placeholder=""
                              />
                              <Button
                                        className=""
                                        disabled={true}
                                        form="createCount"
                                        onClick={[Function]}
                                        type="submit"
                              >
                                        Criar conta
                              </Button>
                    </form>
          </Unknown>
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Logo
              alt="Olist logo"
              className=""
              logo="logo.svg"
/>,
            <Unknown>
              <h1>
                            Crie sua conta
              </h1>
              <form
                            className="form__create-count"
              >
                            <Input
                                          className=""
                                          focus={true}
                                          label="Nome completo"
                                          onChange={[Function]}
                                          placeholder=""
                                          required={true}
                                          type="text"
                            />
                            <Input
                                          className=""
                                          focus={false}
                                          label="E-mail"
                                          onChange={[Function]}
                                          placeholder=""
                                          required={true}
                                          type="text"
                            />
                            <Password
                                          confirmPwd={true}
                                          max={16}
                                          min={6}
                                          passwordValid={[Function]}
                                          placeholder=""
                            />
                            <Button
                                          className=""
                                          disabled={true}
                                          form="createCount"
                                          onClick={[Function]}
                                          type="submit"
                            >
                                          Criar conta
                            </Button>
              </form>
</Unknown>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "Olist logo",
              "className": "",
              "logo": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <h1>
                  Crie sua conta
</h1>,
                <form
                  className="form__create-count"
>
                  <Input
                                    className=""
                                    focus={true}
                                    label="Nome completo"
                                    onChange={[Function]}
                                    placeholder=""
                                    required={true}
                                    type="text"
                  />
                  <Input
                                    className=""
                                    focus={false}
                                    label="E-mail"
                                    onChange={[Function]}
                                    placeholder=""
                                    required={true}
                                    type="text"
                  />
                  <Password
                                    confirmPwd={true}
                                    max={16}
                                    min={6}
                                    passwordValid={[Function]}
                                    placeholder=""
                  />
                  <Button
                                    className=""
                                    disabled={true}
                                    form="createCount"
                                    onClick={[Function]}
                                    type="submit"
                  >
                                    Criar conta
                  </Button>
</form>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Crie sua conta",
                },
                "ref": null,
                "rendered": "Crie sua conta",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Input
                      className=""
                      focus={true}
                      label="Nome completo"
                      onChange={[Function]}
                      placeholder=""
                      required={true}
                      type="text"
/>,
                    <Input
                      className=""
                      focus={false}
                      label="E-mail"
                      onChange={[Function]}
                      placeholder=""
                      required={true}
                      type="text"
/>,
                    <Password
                      confirmPwd={true}
                      max={16}
                      min={6}
                      passwordValid={[Function]}
                      placeholder=""
/>,
                    <Button
                      className=""
                      disabled={true}
                      form="createCount"
                      onClick={[Function]}
                      type="submit"
>
                      Criar conta
</Button>,
                  ],
                  "className": "form__create-count",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "",
                      "focus": true,
                      "label": "Nome completo",
                      "onChange": [Function],
                      "placeholder": "",
                      "required": true,
                      "type": "text",
                    },
                    "ref": "inputRefName",
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "",
                      "focus": false,
                      "label": "E-mail",
                      "onChange": [Function],
                      "placeholder": "",
                      "required": true,
                      "type": "text",
                    },
                    "ref": "inputRefEmail",
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "confirmPwd": true,
                      "max": 16,
                      "min": 6,
                      "passwordValid": [Function],
                      "placeholder": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Criar conta",
                      "className": "",
                      "disabled": true,
                      "form": "createCount",
                      "onClick": [Function],
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Criar conta",
                    "type": [Function],
                  },
                ],
                "type": "form",
              },
            ],
            "type": Symbol(react.fragment),
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
